From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: Use correct flags when reporting status to erp
References: bnc#518291,LTC#54463

Symptom:     The erp might not detect correctly problems from the
             gid_pn nameserver requests.
Problem:     The wrong status flags have been used to pass updates via
             the call to zfcp_erp_notify.
Solution:    Use the correct status flags when calling zfcp_erp_notify.

Acked-by: John Jolly <jjolly@suse.de>
---

 drivers/s390/scsi/zfcp_erp.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/s390/scsi/zfcp_erp.c	2009-07-01 11:31:02.000000000 +0200
+++ b/drivers/s390/scsi/zfcp_erp.c	2009-07-01 12:44:45.000000000 +0200
@@ -850,10 +850,10 @@ void zfcp_erp_port_strategy_open_lookup(
 
 	retval = zfcp_fc_ns_gid_pn(&port->erp_action);
 	if (retval == -ENOMEM)
-		zfcp_erp_notify(&port->erp_action, ZFCP_ERP_NOMEM);
+		zfcp_erp_notify(&port->erp_action, ZFCP_STATUS_ERP_LOWMEM);
 	port->erp_action.step = ZFCP_ERP_STEP_NAMESERVER_LOOKUP;
 	if (retval)
-		zfcp_erp_notify(&port->erp_action, ZFCP_ERP_FAILED);
+		zfcp_erp_notify(&port->erp_action, 0);
 	zfcp_port_put(port);
 }
 
